x/stakeibc/types/message_add_validators.go

```diff
@@ -0,0 +1,69 @@
+package types
+
+import (
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v7/utils"
+)
+
+const TypeMsgAddValidators = "add_validators"
+
+var _ sdk.Msg = &MsgAddValidators{}
+
+func NewMsgAddValidators(creator string, hostZone string, validators []*Validator) *MsgAddValidators {
+	return &MsgAddValidators{
+		Creator:    creator,
+		HostZone:   hostZone,
+		Validators: validators,
+	}
+}
+
+func (msg *MsgAddValidators) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgAddValidators) Type() string {
+	return TypeMsgAddValidators
+}
+
+func (msg *MsgAddValidators) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgAddValidators) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgAddValidators) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+	if err := utils.ValidateAdminAddress(msg.Creator); err != nil {
+		return err
+	}
+
+	if len(msg.Validators) == 0 {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "at least one validator must be provided")
+	}
+
+	for i, validator := range msg.Validators {
```

Comment:
can we validate validator addresses, even though they're on a different chain?

Reply:
maybe also check weights >= 0 here

Reply:
I was messing around with the validator addresses but couldn't really figure out how to do it with the functions at our disposal cause we need each chain's valoper prefix. We could add the valoper prefix to our host zone though - I don't think this is the first time we could have benefitted from it - but maybe that's for a separate PR

Weight is a uint so it has to be >= 0

Reply:
chatted offline - doesn't seem to be a way to validate the validators address without knowing the host zone valoper prefix. We can revisit if we ever start storing the prefix